app/src/main/java/com/example/evently/ui/common/ConfirmFragmentTextInput.java

```diff
@@ -0,0 +1,157 @@
+package com.example.evently.ui.common;
+
+import java.util.function.Predicate;
+
+import android.os.Bundle;
+import android.text.Editable;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.fragment.app.DialogFragment;
+
+import com.google.android.material.textfield.TextInputEditText;
+
+import com.example.evently.R;
+
+/**
+ * Fragment to query user for confirmation with user text input
+ * Connected to fragment_confirmation_text_input.xml
+ */
+public class ConfirmFragmentTextInput extends DialogFragment {
+    final Predicate<String> validate;
+    static final String headerKey = "header";
+    static final String messageKey = "message";
+    static final String cancelKey = "cancel";
+    static final String confirmKey = "confirm";
+    static final String hintKey = "hint";
+    public static final String requestKey = "confirmResult";
+    public static final String inputKey = "input";
+
+    /**
+     * Constructor for ConfirmFragmentTextInput
+     * @param validate Criteria for the string returned.
+     */
+    public ConfirmFragmentTextInput(Predicate<String> validate) {
+        this.validate = validate;
+    }
```

Comment:
This will break when the fragment is recreated during any config changes (happen a lot on Android during app runtime). Functions cannot be arguments to fragments.

Typical way to have an "input function" is to have an abstract fragment with an abstract method, and have implementors implement a fragment for each method. It sucks, but that's unfortunately the way it works.